src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -319,6 +320,7 @@ private unsafe static bool TryIndicesOf(ref byte searchSpace, byte value, int le
                     }
 
                     // No Matches
+                    result = true;
```

Comment:
Yes, result does not mean that anything is found it means that the end was reached